src/main.rs

```diff
@@ -170,16 +172,24 @@ fn sync(lockfile: &Path,
         // Next up, copy it to the vendor directory
         let name = format!("{}-{}", id.name(), id.version());
         let src = src.join(&name).into_path_unlocked();
-        let dst_name = if !explicit_version && id.version() == max[id.name()] {
-            id.name().to_string()
-        } else {
+        let dir_has_version_suffix = explicit_version || id.version() != max[id.name()];
+        let dst_name = if dir_has_version_suffix {
+            // Eg vendor/futures-0.1.13
             format!("{}-{}", id.name(), id.version())
+        } else {
+            // Eg vendor/futures
+            id.name().to_string()
         };
         let dst = local_dst.join(&dst_name);
+
         let cksum = dst.join(".cargo-checksum.json");
-        if cksum.exists() {
+        if dir_has_version_suffix && cksum.exists() {
+            // Always re-copy directory without version suffix in case the version changed
```

Comment:
Nah this is fine, I think this was probably premature optimization on my part which made the tool unnecessarily hard to use!

One use case I'll eventually want to support is to vendor multiple lock files into one directory which may necessitate not blowing away the `vendor` dir, but then again you could also pass multiple arguments...